js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+            ticker = await this.spotPublicGetTicker24hr (this.extend (request, query));
+            //
+            //     {
+            //         "symbol": "BTCUSDT",
+            //         "priceChange": "184.34",
+            //         "priceChangePercent": "0.00400048",
+            //         "prevClosePrice": "46079.37",
+            //         "lastPrice": "46263.71",
+            //         "lastQty": "",
+            //         "bidPrice": "46260.38",
+            //         "bidQty": "",
+            //         "askPrice": "46260.41",
+            //         "askQty": "",
+            //         "openPrice": "46079.37",
+            //         "highPrice": "47550.01",
+            //         "lowPrice": "45555.5",
+            //         "volume": "1732.461487",
+            //         "quoteVolume": null,
+            //         "openTime": 1641349500000,
+            //         "closeTime": 1641349582808,
+            //         "count": null
+            //     }
+            //
+        } else if (marketType === 'swap') {
+            const response = await this.contractPublicGetTicker (this.extend (request, query));
+            //     {
+            //         "success":true,
+            //         "code":0,
+            //         "data":{
+            //             "symbol":"ETH_USDT",
+            //             "lastPrice":3581.3,
+            //             "bid1":3581.25,
+            //             "ask1":3581.5,
+            //             "volume24":4045530,
+            //             "amount24":141331823.5755,
+            //             "holdVol":5832946,
+            //             "lower24Price":3413.4,
+            //             "high24Price":3588.7,
+            //             "riseFallRate":0.0275,
+            //             "riseFallValue":95.95,
+            //             "indexPrice":3580.7852,
+            //             "fairPrice":3581.08,
+            //             "fundingRate":0.000063,
+            //             "maxBidPrice":3938.85,
+            //             "minAskPrice":3222.7,
+            //             "timestamp":1634162885016
+            //         }
+            //     }
+            //
+            ticker = this.safeValue (response, 'data', {});
+        }
+        return this.parseTicker (ticker, market);
+    }
+
+    parseTicker (ticker, market = undefined) {
+        const marketId = this.safeString (ticker, 'symbol');
+        market = this.safeMarket (marketId, market);
+        let timestamp = undefined;
+        let bid = undefined;
+        let ask = undefined;
+        let bidVolume = undefined;
+        let askVolume = undefined;
+        let baseVolume = undefined;
+        let quoteVolume = undefined;
+        let open = undefined;
+        let high = undefined;
+        let low = undefined;
+        let changePcnt = undefined;
+        let changeValue = undefined;
+        let prevClose = undefined;
+        // if swap
+        if ('timestamp' in ticker) {
```

Comment:
This condition is fragile because timestamp is an agnostic property and can be added in the future to spot markets. If you really want to split the spot and swap parsing why not use something like `market['symbol']` it is available for both and it's bulletproof

Reply:
right, but are we sure market is always available? the signature (market = undefiend) probably suggests that we shouldn't rely that market is always defined. let me check again, maybe you are right.

Reply:
i've replaced that with `if (market['swap'] || 'timestamp' in ticker)` . they have recently released the spot-api so they not be going to change it in near future probably. anyway, per your suggestion, if symbol is present, it will not fail 👍

Reply:
@ttodua From what I could test symbol is always available on the response and from that we can always infer the market regardless if it's passed to `parseTicker` or not 🙁

Reply:
> @ttodua From what I could test symbol is always available on the response and from that we can always infer the market regardless if it's passed to `parseTicker` or not 🙁

If the symbol is always available, then we can **almost** always infer the market from it, except for cases when a new market is listed that we don't have in exchange.markets_by_id.